### PR TITLE
Move gatewayHostname param to ClientOptions

### PR DESCRIPTION
**Pull request by drwill-ms**

### CODE REVIEW

File: iothub/device/src/ClientFactory.cs [hunk 1]
```diff
@@ -19,7 +19,7 @@ internal class ClientFactory
 
         private static readonly TimeSpan s_regexTimeoutMilliseconds = TimeSpan.FromMilliseconds(500);
 
-        private static readonly Regex s_deviceIdParameterRegex = new Regex(
+        private static readonly Regex s_deviceIdParameterRegex = new(
```
Comment (timtay-microsoft) [inline iothub/device/src/ClientFactory.cs:22]: I know this is better, but this will never not look funny to me

**Review by timtay-microsoft — APPROVED** [review]

File: iothub/device/src/ClientFactory.cs (hunk 1 above)
Comment (drwill-ms) [reply iothub/device/src/ClientFactory.cs:22]: Not much different than `var foo = new Foo();` but it solves the problem of long lines in class field definitions, which I really appreciate the lack of redundancy. :)

But I hear you on it looking a bit off. ;)

**Pull request merged**